service/resource/deployment/resource.go

```diff
@@ -85,6 +84,9 @@ func New(config Config) (*Resource, error) {
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty.")
 	}
+	if config.TemplateURIVersion == "" {
```

Comment:
Thinking loud. Do we have to ensure a certain format for the version?

Reply:
Do you suggest types like in k8scc? Maybe. This is probably meant to come from the TPO. We'll see. This is also less strict than k8scc. GH URLs may change dynamically as oposed the the content of imported library.

Reply:
I suggest to maybe validate a scheme. I did not talk about types.

Reply:
ATM they are github branch name. Can change to something else in the future, like some blob storage on Azure. I don't want to spend time here now.